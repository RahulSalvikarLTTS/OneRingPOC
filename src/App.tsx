import React from 'react'
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createDrawerNavigator } from '@react-navigation/drawer';
const Drawer = createDrawerNavigator();

// Import files 
import LoginScreen from './Screens/LoginScreen';
import SignUpScreen from './Screens/SignUpScreen';
import DashboardScreen from './Screens/DashboardScreen';

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName='LoginScreen'>
        <Stack.Screen name="LoginScreen" component={LoginScreen} options={{title: "Home", headerShown: false}}/>
        <Stack.Screen name="SignUpScreen" component={SignUpScreen} options={{title: "Sign Up"}} />
        <Stack.Screen name="DashboardScreen" component={DashboardScreen} options={{headerShown: false, headerBackVisible: false, gestureEnabled: false,}}/>
      </Stack.Navigator>
    </NavigationContainer>
  )
}