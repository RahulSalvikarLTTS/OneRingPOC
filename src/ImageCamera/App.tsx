import * as React from 'react';
import { Fragment, useEffect, useState } from 'react'
import { StyleSheet, SafeAreaView, View, Image, ScrollView, Text, Alert } from 'react-native';
import { DemoTitle, DemoButton, DemoResponse } from './components';
import AsyncStorage from "@react-native-async-storage/async-storage";
import * as ImagePicker from 'react-native-image-picker';

import DocumentPicker, {
  DirectoryPickerResponse,
  DocumentPickerResponse,
  isInProgress,
  types,
} from 'react-native-document-picker'
import axios from 'axios';
/* toggle includeExtra */
const includeExtra = true;

export default function App() {
  const [isHideDummyImage, setDummyImageStatus] = useState(false)
  const [FileManagerPath, setFileManagerPath] = useState('')
  const [response, setResponse] = useState<any>(null);
  const [result, setResult] = useState<Array<DocumentPickerResponse> | DirectoryPickerResponse | undefined | null>()

  //------ File Upload

  useEffect(() => {
    //console.log("file manager response -> ",JSON.stringify(result, null, 2));
    //console.log("Image response -> ",JSON.stringify(response, null, 2));
    let data = JSON.stringify(result, null, 2);
    // if ( != undefined) {
    // setFileManagerPath(result[0].fileCopyUri)
    // }
  }, [result, response])

  const handleError = (err: unknown) => {
    if (DocumentPicker.isCancel(err)) {
      console.warn('cancelled')
      // User cancelled the picker, exit any dialogs or menus and move on
    } else if (isInProgress(err)) {
      console.warn('multiple pickers were opened, only the last will be considered')
    } else {
      throw err
    }
  }
  //------
  const setDefult = () => {
    setDummyImageStatus(false)
    setResponse(null)
    setResult(null)
    setFileManagerPath('')
  }

  const onButtonPress = React.useCallback((type, options) => {
    if (type === 'capture') {
      setDefult()
      cameraAction(options);
      //ImagePicker.launchCamera(options, setResponse);
    } else if (type === 'library') {
      setDefult()
      galleryAction(options);
      //ImagePicker.launchImageLibrary(options, setResponse);
    } else if (type === 'FileManager') {
      setDefult()
      fileManagerAction(options)
    } else {
      uploadAction()
    }
  }, []);


  const fileManagerAction = async (options: any) => {
    try {
      const pickerResult = await DocumentPicker.pickSingle({
        presentationStyle: 'fullScreen',
        copyTo: 'cachesDirectory',
      })
      //console.log("pickerResult ->" ,pickerResult.fileCopyUri);
      await AsyncStorage.setItem('fileUri', String(pickerResult.fileCopyUri));
      setFileManagerPath(String(pickerResult.fileCopyUri))
      setResult([pickerResult]);
    } catch (e) {
      handleError(e)
    }

  }

  const galleryAction = async (options: any) => {
    try {
      const result = await ImagePicker.launchImageLibrary(options);
      await AsyncStorage.setItem('fileUri', String(result?.assets[0].uri));
      setResponse(result);
      setDummyImageStatus(true)
    } catch (e) {
      handleError(e)
    }
  }


  const cameraAction = async (options: any) => {
    try {
      const result = await ImagePicker.launchCamera(options);
      await AsyncStorage.setItem('fileUri', String(result?.assets[0].uri));
      setResponse(result);
      setDummyImageStatus(true)
    } catch (e) {
      handleError(e)
    }
  }

  const uploadAction = () => {
    try {
      let fileUrl = ''
      AsyncStorage.getItem('fileUri')
        .then(value => {
          if (value != null) {
            fileUrl = value
            console.log("value", value)
            imageUpload(value)
          }
        })

    } catch (error) {
      console.log("Get Data error ->", error);
    }
  }
  /*
  const imageUpload = (fileUri: String) => {
    console.log("fileUri ", fileUri);
    handleSubmit(fileUri);
    //AsyncStorage.removeItem('fileUri')

    // const imageData = new FormData()
    // imageData.append("file", {
    //   uri: fileUri,
    //   name: 'Img1.png',
    //   fileName: 'image',
    //   type: 'image/png'
    // })
    // console.log('form data', imageData)
    // axios({
    //   method: 'post',
    //   uri: '',
    //   data: imageData
    // }).then(function (response){
    //   console.log("image upload success", response.data)
    // }).then((error) {
    //   console.log(error)
    // })
  }*/

  const imageUpload = async(fileUri: String) => {
    const formData = new FormData();
    var filename = fileUri.substring(fileUri.lastIndexOf('/')+1);
    
    formData.append("file", {
      uri: fileUri,
      name: filename,
      fileName: filename,
      type: 'image/png'
    })
    formData.append("file", fileUri);
    try {
      const response = await axios({
        method: "post",
        url: "https://api.upload.io/v2/accounts/kW15b68/uploads/form_data",
        data: formData,
        headers: { "Authorization": "Bearer public_kW15b682j4mHVWyJR9tLvL34kwQh"},
      }).then(function(response){
        console.log("image upload success", response.data)
        Alert.alert('Success', 'File Uploaded Successfully');
      }).then((error) => {
        console.log(error)
        Alert.alert('Failed', 'File Failed to Uploaded');
      })
    } catch(error) {
      console.log(error)
      Alert.alert('Alert', 'No Internet Connection');
    } 
  }

  const renderFileUri = () => {
    if (isHideDummyImage == false) {
      return <View style={styles.imageContainer}>
        <Image
          source={require('./assets/dummy.png')}
          style={styles.image}
        />
        <Text style={styles.text}>File path: {FileManagerPath}</Text>

      </View>

    }
  }


  return (
    <Fragment>
      <SafeAreaView style={styles.container}>
        <DemoTitle>Honeywell</DemoTitle>
        {/* <DemoResponse>{response}</DemoResponse> */}

        <ScrollView>

          {response?.assets &&
            response?.assets.map(({ uri }: { uri: string }) => (
              <View key={uri} style={styles.imageContainer}>
                <Image
                  resizeMode="cover"
                  resizeMethod="scale"
                  style={styles.image}
                  source={{ uri: uri }}
                />

                <Text style={styles.text}>File path: {uri}</Text>

              </View>
            ))}

          {renderFileUri()}

          <View style={styles.buttonContainer}>
            {actions.map(({ title, type, options }) => {
              return (
                <DemoButton
                  key={title}
                  onPress={() => onButtonPress(type, options)}>
                  {title}
                </DemoButton>
              );
            })}
          </View>

        </ScrollView>
      </SafeAreaView>
    </Fragment>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'alicered',
  },
  buttonContainer: {
    flexDirection: 'column',
    //flexWrap: 'wrap',
    marginVertical: 8,
    alignItems: 'center'
  },
  imageContainer: {
    marginVertical: 24,
    alignItems: 'center',
  },
  image: {
    width: 200,
    height: 200,
    alignItems: 'center',
  },
  text: {
    alignItems: 'center',
    padding: 20,
  },
});

interface Action {
  title: string;
  type: 'capture' | 'library' | 'FileManager' | 'Upload';
  options: ImagePicker.CameraOptions | ImagePicker.ImageLibraryOptions;
}

const actions: Action[] = [
  {
    title: 'Camera',
    type: 'capture',
    options: {
      saveToPhotos: true,
      mediaType: 'photo',
      includeBase64: false,
      includeExtra,
    },
  },
  {
    title: 'Gallery',
    type: 'library',
    options: {
      selectionLimit: 0,
      mediaType: 'photo',
      includeBase64: false,
      includeExtra,
    },
  },
  {
    title: 'File Manager',
    type: 'FileManager',
    options: {
      selectionLimit: 0,
      mediaType: 'photo',
      includeBase64: false,
      includeExtra,
    },
  },
  {
    title: 'Upload',
    type: 'Upload',
    options: {
      selectionLimit: 0,
      mediaType: 'photo',
      includeBase64: false,
      includeExtra,
    },
  },
];
