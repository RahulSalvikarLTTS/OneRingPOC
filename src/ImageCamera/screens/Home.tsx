import { View, Text, Image, Button } from 'react-native'
import React, { Children, Fragment, useEffect, useState } from 'react'
import Styles from './Styles'
import DummyImageView from './atom/DummyImageView'
import FileNameText from './atom/FileNameText'
import { DemoButton } from '../components'
import * as ImagePicker from 'react-native-image-picker';
import { useSelector, useDispatch } from "react-redux";
import { gallery, camera, fileManager } from "../redux/action";
import * as DocumentPicker from 'react-native-document-picker'
/* toggle includeExtra */
const includeExtra = true;

export default function Home() {
    const filePath = useSelector((state: any) => state.filePath)
    const dispatch = useDispatch();
    const onButtonPress = React.useCallback((type: any, options: any) => {
        if (type === 'capture') {
            cameraAction(options);
        } else if (type === 'library') {
            galleryAction(options);
        } else if (type === 'FileManager') {
            fileManagerAction(options)
        } else {
            console.log('newPath ->', filePath)
        }
    }, [filePath]);

    const cameraAction = async (options: any) => {
        try {
            const result = await ImagePicker.launchCamera(options);
            dispatch(camera(result))
        } catch (error) {
            console.log(error)
        }
    }

    const galleryAction = async (options: any) => {
        try {
            const result = await ImagePicker.launchImageLibrary(options);
            dispatch(gallery(result))
        } catch (error) {
            console.log(error)
        }
    }

    const fileManagerAction = async (options: any) => {
        try {
            const pickerResult = await DocumentPicker.pickSingle({
                presentationStyle: 'fullScreen',
                copyTo: 'cachesDirectory',
            })
            dispatch(fileManager(pickerResult))
        } catch (error) {
            console.log(error)
        }

    }

    const renderFileUri = () => {
          return <View style={Styles.imageContainer}>
            <Image
            source={filePath == '' ? require('../assets/dummy.png') : { uri: filePath }}
            style={Styles.image}
          />
          </View>
    
      }
    
    return (
        <View style={Styles.container}>
            <View style={Styles.imageTextContainer}>
            {/* {renderFileUri()} */}
                <DummyImageView imageUrl={filePath} /> 
                <FileNameText>{filePath}</FileNameText>
            </View>
            <View style={Styles.buttonContainer}>
                {actions.map(({ title, type, options }) => {
                    return (
                        <DemoButton
                            key={title}
                            onPress={() => onButtonPress(type, options)}>
                            {title}
                        </DemoButton>
                    );
                })}
            </View>
        </View>
    )
}

interface Action {
    title: string;
    type: 'capture' | 'library' | 'FileManager' | 'Upload';
    options: ImagePicker.CameraOptions | ImagePicker.ImageLibraryOptions | DocumentPicker.DocumentPickerResponse | null;
}

const actions: Action[] = [
    {
        title: 'Camera',
        type: 'capture',
        options: {
            saveToPhotos: true,
            mediaType: 'photo',
            includeBase64: false,
            includeExtra,
        },
    },
    {
        title: 'Gallery',
        type: 'library',
        options: {
            selectionLimit: 0,
            mediaType: 'photo',
            includeBase64: false,
            includeExtra,
        },
    },
    {
        title: 'File Manager',
        type: 'FileManager',
        options: {
            selectionLimit: 0,
            mediaType: 'photo',
            includeBase64: false,
            includeExtra,
        },
    },
    {
        title: 'Upload',
        type: 'Upload',
        options: null,
    },
];
