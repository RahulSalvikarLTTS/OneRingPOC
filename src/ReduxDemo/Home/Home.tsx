import { Text, StyleSheet, View, Button } from 'react-native'
import React, { useEffect, useState } from 'react'
import store from '../redux/store'
import { decrement, increment } from '../redux/action'


export default function Home() {
  const [number, setNumber] = useState(0)

  useEffect(() => {
    const unsubscribe = store.subscribe(() => {
      let value = store.getState().num
      setNumber(value)
    })
  
    return () => {
      unsubscribe()
    }
  }, [])
  

  const onAdd = () => {
    store.dispatch(increment(number))
  }
  const onMinus = () => {
    store.dispatch(decrement(number))
  }

  return (
    <View style={styles.container}>
      <Text>{number}</Text>
      <Button title='Add' onPress={onAdd}></Button>
      <Button title='Minus' onPress={onMinus}></Button>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white'
  }
})