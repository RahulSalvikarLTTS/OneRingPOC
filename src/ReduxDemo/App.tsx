import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";
import { Provider } from "react-redux";
import store from "./redux/store";
import Home from "./screens/home";

const App = () => {
  return (
    <Provider store={store}>
      <Home />
    </Provider>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
})

export default App;