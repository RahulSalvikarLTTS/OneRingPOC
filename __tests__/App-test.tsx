/**
 * @format
 */

import 'react-native';
import React from 'react';
//import App from '../src/LoginDemo/App';
import App from '../src/ImageCamera/App';
import Home from '../src/ImageCamera/screens/Home';

//import App from '../src/App';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// it('renders correctly', () => {
//   //renderer.create(<App />).toJSON();
//   const snap = renderer.create(
//     <App />
//   ).toJSON();
//   expect(snap).toMatchSnapshot();
// });

if ("check function", () => {
  let HomeData = renderer.create(<Home />).getInstance();
  console.log(HomeData);
})