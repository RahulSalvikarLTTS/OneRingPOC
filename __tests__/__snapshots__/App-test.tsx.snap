// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Honeywell"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 50,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../src/ImageCamera/assets/dummy.png",
                  }
                }
                style={
                  {
                    "alignItems": "center",
                    "height": 200,
                    "width": 200,
                  }
                }
              />
              <Text />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#5A5A5A",
                    },
                    {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flex": 1,
                      "height": 50,
                      "justifyContent": "center",
                      "marginHorizontal": 8,
                      "marginVertical": 10,
                      "maxWidth": "100%",
                      "minWidth": "80%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                      "textAlign": "center",
                    }
                  }
                >
                  Camera
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#5A5A5A",
                    },
                    {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flex": 1,
                      "height": 50,
                      "justifyContent": "center",
                      "marginHorizontal": 8,
                      "marginVertical": 10,
                      "maxWidth": "100%",
                      "minWidth": "80%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                      "textAlign": "center",
                    }
                  }
                >
                  Gallery
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#5A5A5A",
                    },
                    {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flex": 1,
                      "height": 50,
                      "justifyContent": "center",
                      "marginHorizontal": 8,
                      "marginVertical": 10,
                      "maxWidth": "100%",
                      "minWidth": "80%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                      "textAlign": "center",
                    }
                  }
                >
                  File Manager
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "backgroundColor": "#5A5A5A",
                    },
                    {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "flex": 1,
                      "height": 50,
                      "justifyContent": "center",
                      "marginHorizontal": 8,
                      "marginVertical": 10,
                      "maxWidth": "100%",
                      "minWidth": "80%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                      "textAlign": "center",
                    }
                  }
                >
                  Upload
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
